--- gcc-4.0.1/gcc/ada/Makefile.in.ada-makefile	2005-06-02 10:33:53.000000000 -0400
+++ gcc-4.0.1/gcc/ada/Makefile.in	2005-06-02 10:35:38.000000000 -0400
@@ -139,7 +139,7 @@ OUTPUT_OPTION = @OUTPUT_OPTION@
 objext = .o
 exeext =
 arext  = .a
-soext  = .so
+soext  = .so.1
 shext  =
 hyphen = -
 
@@ -1939,6 +1939,10 @@ gnatlib-shared-default:
 		libgnat$(soext)
 	cd rts; $(LN_S) libgnarl$(hyphen)$(LIBRARY_VERSION)$(soext) \
 		libgnarl$(soext)
+	cd rts; $(LN_S) libgnat$(hyphen)$(LIBRARY_VERSION)$(soext) \
+		libgnat$(hyphen)$(LIBRARY_VERSION).so
+	cd rts; $(LN_S) libgnarl$(hyphen)$(LIBRARY_VERSION)$(soext) \
+		libgnarl$(hyphen)$(LIBRARY_VERSION).so
 
 gnatlib-shared-dual:
 	$(MAKE) $(FLAGS_TO_PASS) \
@@ -1946,14 +1950,14 @@ gnatlib-shared-dual:
 	     GNATLIBCFLAGS="$(GNATLIBCFLAGS)" \
 	     THREAD_KIND="$(THREAD_KIND)" \
              gnatlib-shared-default
-	$(MV) rts/libgna*$(soext) .
+	$(MV) rts/libgna*$(soext) rts/libgna*.so .
 	$(RM) ../stamp-gnatlib2
 	$(MAKE) $(FLAGS_TO_PASS) \
              GNATLIBFLAGS="$(GNATLIBFLAGS)" \
 	     GNATLIBCFLAGS="$(GNATLIBCFLAGS)" \
 	     THREAD_KIND="$(THREAD_KIND)" \
              gnatlib
-	$(MV) libgna*$(soext) rts
+	$(MV) libgna*$(soext) libgna*.so rts
 
 gnatlib-shared-dual-win32:
 	$(MAKE) $(FLAGS_TO_PASS) \
